semantic_router/index/qdrant.py

```diff
@@ -0,0 +1,226 @@
+    )
+    prefix: Optional[str] = Field(
+        default=None,
+        description="Prefix to the REST URL path. Example: `http://localhost:6333/some/prefix/{qdrant-endpoint}`.",
+    )
+    timeout: Optional[int] = Field(
+        default=None,
+        description="Timeout for REST and gRPC API requests.",
+    )
+    host: Optional[str] = Field(
+        default=None,
+        description="Host name of Qdrant service. If url and host are None, set to 'localhost'.",
+    )
+    path: Optional[str] = Field(
+        default=None,
+        description="Persistence path for Qdrant local",
+    )
+    grpc_options: Optional[Dict[str, Any]] = Field(
+        default=None,
+        description="Options to be passed to the low-level Qdrant GRPC client, if used.",
+    )
+    size: Union[int, None] = Field(
+        default=None,
+        description="Embedding dimensions. Defaults to the embedding length of the configured encoder.",
+    )
+    distance: str = Field(
+        default="Cosine", description="Distance metric to use for similarity search."
+    )
+    collection_options: Optional[Dict[str, Any]] = Field(
+        default={},
+        description="Additonal options to be passed to `QdrantClient#create_collection`.",
+    )
+    client: Any = Field(default=None, exclude=True)
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self.type = "qdrant"
+        self.client = self._initialize_client()
+
+    def _initialize_client(self):
+        try:
+            from qdrant_client import QdrantClient
+
+            return QdrantClient(
+                location=self.location,
+                url=self.url,
+                port=self.port,
+                grpc_port=self.grpc_port,
+                prefer_grpc=self.prefer_grpc,
+                https=self.https,
+                api_key=self.api_key,
+                prefix=self.prefix,
+                timeout=self.timeout,
+                host=self.host,
+                path=self.path,
+                grpc_options=self.grpc_options,
+            )
+
+        except ImportError as e:
+            raise ImportError(
+                "Please install 'qdrant-client' to use QdrantIndex."
+                "You can install it with: "
+                "`pip install 'semantic-router[qdrant]'`"
+            ) from e
+
+    def _init_collection(self) -> None:
+        from qdrant_client import QdrantClient, models
+
+        self.client: QdrantClient
+        if not self.client.collection_exists(self.collection_name):
+            if not self.dimensions:
+                raise ValueError(
+                    "Cannot create a collection without specifying the dimensions."
+                )
+
+            self.client.create_collection(
+                collection_name=self.collection_name,
+                vectors_config=models.VectorParams(
+                    size=self.dimensions,
+                    distance=self.distance,  # type: ignore
+                ),
```

Comment:
Given that qdrant has slightly different metrics to the other indexes, I think we need a `Metric` type in `schema.py` which will include standard string values that will be used by the library and converted into the correct metric string for each service. Something like:

```python
class Metric(Enum):
    COSINE = "cosine"
    DOTPRODUCT = "dotproduct"
    EUCLIDEAN = "euclidean"
    MANHATTEN = "manhattan"
```

Reply:
https://github.com/aurelio-labs/semantic-router/pull/207/commits/a9bcc0ee2c56f02cca82e94f0dd9cdb498db081f

Reply:
https://github.com/aurelio-labs/semantic-router/pull/207/commits/78a2daec28663d2025d7d6cf40e44300786f58b6